editor/script_create_dialog.cpp

```diff
@@ -112,48 +112,23 @@ static Vector<String> _get_hierarchy(const String &p_class_name) {
 void ScriptCreateDialog::_notification(int p_what) {
 	switch (p_what) {
 		case NOTIFICATION_ENTER_TREE: {
-			String last_language = EditorSettings::get_singleton()->get_project_metadata("script_setup", "last_selected_language", "");
-			if (!last_language.is_empty()) {
-				for (int i = 0; i < language_menu->get_item_count(); i++) {
-					if (language_menu->get_item_text(i) == last_language) {
-						language_menu->select(i);
-						break;
-					}
-				}
-			} else {
-				language_menu->select(default_language);
-			}
+			_update_languages();
 			is_using_templates = EDITOR_DEF("_script_setup_use_script_templates", false);
 			use_templates->set_pressed(is_using_templates);
 		} break;
 
 		case NOTIFICATION_THEME_CHANGED: {
-			const int icon_size = get_theme_constant(SNAME("class_icon_size"), EditorStringName(Editor));
-
-			EditorData &ed = EditorNode::get_editor_data();
-
-			for (int i = 0; i < ScriptServer::get_language_count(); i++) {
-				// Check if the extension has an icon first.
-				String script_type = ScriptServer::get_language(i)->get_type();
-				Ref<Texture2D> language_icon = get_editor_theme_icon(script_type);
-				if (language_icon.is_null() || language_icon == ThemeDB::get_singleton()->get_fallback_icon()) {
-					// The theme doesn't have an icon for this language, ask the extensions.
-					Ref<Texture2D> extension_language_icon = ed.extension_class_get_icon(script_type);
-					if (extension_language_icon.is_valid()) {
-						language_menu->get_popup()->set_item_icon_max_width(i, icon_size);
-						language_icon = extension_language_icon;
-					}
-				}
-
-				if (language_icon.is_valid()) {
-					language_menu->set_item_icon(i, language_icon);
-				}
-			}
-
+			_update_languages();
 			path_button->set_button_icon(get_editor_theme_icon(SNAME("Folder")));
 			parent_browse_button->set_button_icon(get_editor_theme_icon(SNAME("Folder")));
 			parent_search_button->set_button_icon(get_editor_theme_icon(SNAME("ClassList")));
 		} break;
+
+		case NOTIFICATION_VISIBILITY_CHANGED: {
+			if (is_visible()) {
+				_update_languages();
+			}
```

Comment:
Instead of here, I think it needs to update before `_language_changed` is called in `ScriptCreateDialog::config`.